dashboard/app/models/skill.rb

```diff
@@ -17,7 +17,9 @@
 class Skill < ApplicationRecord
   validates :description, presence: true
 
-  has_and_belongs_to_many :levels, join_table: 'levels_skills'
+  has_and_belongs_to_many :levels, join_table: 'levels_skills', dependent: :delete_all
+
+  after_destroy :delete_serialized_file
```

Comment:
It looks like our [normal pattern](https://github.com/code-dot-org/code-dot-org/blob/be1f80adfdce795c531cb07f3879144cd6d07f2b/dashboard/app/models/programming_class.rb#L35) is to do this `after_destroy`